File: microservers/billetterieservice/src/main/java/io/multi/billetterieservice/domain/Ville.java
package io.multi.billetterieservice.domain;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Ville {

    private Long villeId;
    private String villeUuid;
    private Long regionId;
    private String libelle;
    private String codePostal;
    private Boolean actif;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;

    // Champs additionnels pour les jointures
    private String regionLibelle;
    private String regionUuid;
}

File: frontend/ultima-ng-19.0.0/src/app/interface/ville.ts
export interface IVille {
    villeId: number;
    villeUuid: string;
    libelle: string;
    codePostal: string;
    actif: boolean;
    regionId: number;
    regionUuid: string;
    regionLibelle: string;
    createdAt: Date | string;
    updatedAt: Date | string;
}

export interface IVilleCreateRequest {
    regionUuid: string;
    libelle: string;
    codePostal?: string;
}

export interface IVilleUpdateRequest {
    libelle: string;
    codePostal?: string;
    regionUuid?: string;
}
